cli/src/description_util.rs

```diff
@@ -107,6 +108,41 @@ pub fn description_template_for_commit(
     }
 }
 
+pub fn description_template_for_commit(
+    ui: &Ui,
+    settings: &UserSettings,
+    workspace_command: &WorkspaceCommandHelper,
+    intro: &str,
+    overall_commit_description: &str,
+    from_tree: &MergedTree,
+    to_tree: &MergedTree,
+) -> Result<String, CommandError> {
+    let mut diff_summary_bytes = Vec::new();
+    diff_util::show_diff(
+        ui,
+        &mut PlainTextFormatter::new(&mut diff_summary_bytes),
+        workspace_command,
+        from_tree,
+        to_tree,
+        &EverythingMatcher,
+        &[DiffFormat::Summary],
+    )?;
+    let mut template_chunks = Vec::new();
+    if !intro.is_empty() {
+        template_chunks.push(format!("JJ: {intro}\n"));
+    }
+    template_chunks.push(if overall_commit_description.is_empty() {
+        settings.default_description()
+    } else {
+        overall_commit_description.to_owned()
+    });
+    if !diff_summary_bytes.is_empty() {
+        template_chunks.push("\n".to_owned());
+        template_chunks.push(diff_summary_to_description(&diff_summary_bytes));
+    }
+    Ok(template_chunks.join(""))
```

Comment:
nit: `.concat()` also works.

Alternatively, the string can be built by using `+=` or `.push_str()`.

Reply:
Thanks, I changed it to `concat()`.